lib/rules/color-no-out-gamut-range/index.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const validateOptions = require('../../utils/validateOptions');
+const ruleMessages = require('../../utils/ruleMessages');
+const valueParser = require('postcss-value-parser');
+// @ts-expect-error -- TS7016
+const Color = require('colorjs.io').default;
+const report = require('../../utils/report');
+const getDeclarationValue = require('../../utils/getDeclarationValue');
+const { isAtRule } = require('../../utils/typeGuards');
+const declarationValueIndex = require('../../utils/declarationValueIndex');
+
+const ruleName = 'color-no-out-gamut-range';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (color) => `Unexpected color out of gamut range "${color}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/color-no-out-gamut-range',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+		});
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls((decl) => {
+			const parsedValue = valueParser(getDeclarationValue(decl));
+
+			// ignore css variable declaration
+			if (decl.prop.startsWith('--')) return;
+
+			// ignore scss variable declaration
+			if (decl.prop.startsWith('$')) return;
+
+			parsedValue.walk((node) => {
+				if (node.type !== 'function') return;
+
+				if (
+					node.value !== 'lch' &&
+					node.value !== 'lab' &&
+					node.value !== 'oklch' &&
+					node.value !== 'oklab'
+				)
+					return;
+
+				const isInSrgbGamut = new Color(node.value, getColorCoordinates(node)).inGamut('srgb');
```

Comment:
[suggest] I noticed the `Color` constructor allows a string argument according to the [color.js doc](https://github.com/LeaVerou/color.js#reading-colors). Does it make sense to simplify the code like this?

```suggestion
				const isInSrgbGamut = new Color(valueParser.stringify(nodes)).inGamut('srgb');
```

```diff
+	parsedValue.walk((node) => {
-	parsedValue.walk((node, _index, nodes) => {
```